segmentstore/server/src/main/java/io/pravega/segmentstore/server/tables/ContainerTableExtensionImpl.java

```diff
@@ -320,9 +332,15 @@ protected int getMaxCompactionSize() {
     }
 
     private <T> CompletableFuture<Long> commit(Collection<T> toCommit, Function<Collection<T>, BufferView> serializer,
-                                               DirectSegmentAccess segment, Duration timeout) {
-        BufferView s = serializer.apply(toCommit);
-        return segment.append(s, null, timeout);
+                                               DirectSegmentAccess segment, long tableSegmentOffset, Duration timeout) {
+        if (tableSegmentOffset == NULL_TABLE_SEGMENT_OFFSET || tableSegmentOffset == segment.getInfo().getLength()) {
```

Comment:
Remove the second part of this check. You cannot rely on the length of the segment at this point as there may be multiple operations in flight that will affect it before your operation is processed.